webserver/src/components/PDBFetch/PDBFetch.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+
+export default function PDBFetch({PDBCode, setPDBCode, submitPressed}) {
+
+    const [pdb, setPDB] = useState("")
+
+    useEffect( () => { 
+        if (pdb.length != 4) { return }
+        if (!pdb.match(/^[a-z0-9]+$/i)) { return }
```

Comment:
I think we should avoid relying on regex to check if a PDB code is valid. Instead, I think it would be a better idea to query PDB API endpoint to check if the PDB code exists ( https://data.rcsb.org/#rest-api )

For example, if I use 9fjj, the PDB code is valid, but has not yet been assigned to a structure on wwPDB: https://[data.rcsb.org/rest/v1/core/entry/9ffj](https://data.rcsb.org/rest/v1/core/entry/9ffj)

Reply:
This is just a pre-filter to stop people from submitting anything that isn't a valid PDB. After this, it checks the ebi API and if it fails to find anything there it shows an error screen.